### PR TITLE
extract deposit percentage from shortcode, pass it in bet params

### PR DESCRIPTION
**Pull request by raunakkathuria**

### CODE REVIEW

File: lib/Finance/Contract/Longcode.pm [hunk 1]
```diff
@@ -187,12 +187,12 @@ sub shortcode_to_parameters {
             $tick_expiry    = 1;
             $how_many_ticks = $5;
         }
-    } elsif ($shortcode =~ /^BINARYICO_(\d+\.?\d*)_(\d+)$/) {
-        $bet_type                      = 'BINARYICO';
-        $underlying_symbol             = 'BINARYICO';
-        $binaryico_per_token_bid_price = $1;
-        $binaryico_number_of_tokens    = $2;
-
+    } elsif ($shortcode =~ /^BINARYICO_(\d+\.?\d*)_(\d+)(?:_(\d))?$/) {
```
Comment (aminmarashi) [inline lib/Finance/Contract/Longcode.pm]: We expect the third match to be one digit, is that expected?

Comment (raunakkathuria) [reply lib/Finance/Contract/Longcode.pm]: thanks will change it

File: lib/Finance/Contract/Longcode.pm [hunk 2]
```diff
@@ -236,10 +236,11 @@ sub shortcode_to_parameters {
...
     # ICO
     if ($bet_type eq 'BINARYICO') {
-        $bet_parameters->{amount_type}                   = 'stake';
-        $bet_parameters->{amount}                        = $binaryico_per_token_bid_price;
-        $bet_parameters->{binaryico_number_of_tokens}    = $binaryico_number_of_tokens;
-        $bet_parameters->{binaryico_per_token_bid_price} = $binaryico_per_token_bid_price;
+        $bet_parameters->{amount_type}                           = 'stake';
+        $bet_parameters->{amount}                                = $binaryico_per_token_bid_deposit_price;
+        $bet_parameters->{binaryico_number_of_tokens}            = $binaryico_number_of_tokens;
+        $bet_parameters->{binaryico_per_token_bid_deposit_price} = $binaryico_per_token_bid_deposit_price;
```
Comment (aminmarashi) [inline lib/Finance/Contract/Longcode.pm]: Why adding `binaryico_deposit_percentage`? Wasn't more efficient to calculate `binaryico_per_token_bid_deposit_price` here?

Comment (aminmarashi) [inline lib/Finance/Contract/Longcode.pm]: Why adding `binaryico_deposit_percentage`, wasn't easier to calculate `binaryico_per_token_bid_deposit_price` based on percentage field here?

Comment (raunakkathuria) [reply lib/Finance/Contract/Longcode.pm]: cos we map shortcode to params, so BINARYICO_1.35_25_5 is mapped to different bet params, calculation specific to how we are using it should not be inside this module

**Pull request closed without merging**